pkg/api/frontend_logging.go

```diff
@@ -28,14 +66,121 @@ type frontendSentryEvent struct {
 	Exception *frontendSentryException `json:"exception,omitempty"`
 }
 
+func guessSourceMapLocation(sourceUrl string) (*sourceMapLocation, error) {
+	u, err := url.Parse(sourceUrl)
+	if err != nil {
+		return nil, err
+	}
+	if strings.HasPrefix(u.Path, "/public/build/") {
+		return &sourceMapLocation{
+			dir:      setting.StaticRootPath,
+			path:     path.Join("build", u.Path[len("/public/build/"):]) + ".map",
+			pluginId: "",
+		}, nil
+	} else if strings.HasPrefix(u.Path, "/public/plugins/") {
+		for _, route := range plugins.StaticRoutes {
+			pluginPrefix := path.Join("/public/plugins/", route.PluginId)
+			if strings.HasPrefix(u.Path, pluginPrefix) {
+				return &sourceMapLocation{
+					dir:      route.Directory,
+					path:     u.Path[len(pluginPrefix):] + ".map",
+					pluginId: route.PluginId,
+				}, nil
+			}
+		}
+	}
+	return nil, nil
+}
+
+func getSourceMap(sourceUrl string) (*sourceMap, error) {
+	sourceMapCache.Lock()
+	defer sourceMapCache.Unlock()
+
+	if smap, ok := sourceMapCache.cache[sourceUrl]; ok {
+		return smap, nil
+	}
+	sourceMapLocation, err := guessSourceMapLocation(sourceUrl)
+	if err != nil {
+		return nil, err
+	}
+	if sourceMapLocation != nil {
```

Comment:
It's more idiomatic to return early here, and avoid nesting. Also wondering if you actually want to delete the cache entry?
```suggestion
	if sourceMapLocation == nil {
		delete(sourceMapCache.cache, sourceURL)
		return nil, nil
	}
```

Reply:
If there's no cache entry, and sourcemap is not found on the fs, I want to add cache entry with `nil` value so we don't try to find it in fs again later

Reply:
OK thanks, updated my suggestion